nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/PutMongoBulk.java

```diff
@@ -0,0 +1,353 @@
+                } else if ("replaceOne".equals(updateType)) {
+                    final ReplaceOptions options = parseReplaceOptions(updateSpec);
+                    final ReplaceOneModel<Document> replaceOneModel = new ReplaceOneModel<>((BsonDocument) updateSpec.get("filter"),
+                            toBsonDocument((BsonDocument) updateSpec.get("replacement")), options);
+                    updateModels.add(replaceOneModel);
+                } else if ("deleteOne".equals(updateType)) {
+                    final DeleteOptions options = parseDeleteOptions(updateSpec);
+                    final DeleteOneModel<Document> deleteOneModel = new DeleteOneModel<>((BsonDocument) updateSpec.get("filter"), options);
+                    updateModels.add(deleteOneModel);
+                } else  if ("deleteMany".equals(updateType)) {
+                    final DeleteOptions options = parseDeleteOptions(updateSpec);
+                    final DeleteManyModel<Document> deleteManyModel = new DeleteManyModel<>((BsonDocument) updateSpec.get("filter"), options);
+                    updateModels.add(deleteManyModel);
+                } else {
+                    logger.error("Invalid bulk-update: invalid update type {}", updateType);
+                    session.transfer(flowFile, REL_FAILURE);
+                    context.yield();
+                    return;
+                }
+            }
+
+            if (context.getProperty(USE_TRANSACTION).asBoolean()) {
+                MongoClient currentMongoClient;
+                if (clientService != null) {
+                    // (can still add - if it's ever needed)
+                    throw new RuntimeException("Cannot have transactions and use a client service - so far");
+                } else {
+                    currentMongoClient = this.mongoClient;
+                }
+                clientSession = currentMongoClient.startSession();
+                clientSession.startTransaction();
+                // now run this w/in a transaction
+                collection.bulkWrite(clientSession, updateModels, (new BulkWriteOptions().ordered(context.getProperty(ORDERED).asBoolean())));
+            } else {
+                collection.bulkWrite(updateModels, (new BulkWriteOptions().ordered(context.getProperty(ORDERED).asBoolean())));
+            }
+            logger.info("bulk-updated {} into MongoDB", flowFile);
+            // (could also return the result again as JSON - mostly not needed afaik)
+
+            session.getProvenanceReporter().send(flowFile, getURI(context));
+            session.transfer(flowFile, REL_SUCCESS);
+
+            if (clientSession != null) {
+                if (clientSession.hasActiveTransaction()) {
+                    clientSession.commitTransaction();
+                }
+                clientSession.close();
+            }
+        } catch (Exception e) {
+            logger.error("Failed to bulk-update {} into MongoDB due to {}", new Object[] {flowFile, e}, e);
```

Comment:
The `Object[]` array wrapper is not necessary, and the `due to {}` convention is not necessary either, so it should be removed.
```suggestion
            logger.error("Failed to bulk-update {} into MongoDB", flowFile, e);
```